src/Uno.UI.Runtime.Skia.Gtk/Input/GtkCorePointerInputSource.cs

```diff
@@ -596,4 +628,23 @@ private void RaisePointerWheelChanged(PointerEventArgs ptArgs, [CallerMemberName
 
 		PointerWheelChanged?.Invoke(this, ptArgs);
 	}
+
+	private static (int x, int y) GetWindowCoordinates(object eventSource)
+	{
+		// Calling GetGeometry on any window reference such as e.Window or (o as Widget).window will return wrong
+		// values (usually just 0). Only UnoGtkWindow.Window will give us what we need.
+		while (eventSource is not UnoGtkWindow)
```

Comment:
I suspect this is a regression from recent changes (GTKHost init? TextBox?). Were we able to determine which changes introduced this? 

I would try to avoid, on each pointer event, to walk the tree up to the `UnoGtkWindow` (BTW not sure we do have such element in XAML islands, should we instead search for `IGtkXamlRootHost` ? @MartinZikmund). Is there a way to cache the relative position? Like we are hooking handlers on a private `EventBox`, could we just capture those coordinates in constructor?

Reply:
It would still be interesting to know if it's a regression. Can you try to go back in preview releases to see if this was changed? This would indicate some additional behavior that got changed.

Reply:
Also, are we sure to always have a `UnoGtkWindow`? Even in XAML islands?

Reply:
Hmm, for that question - yes and no - in GTK we have no islands (yet) - but if added sometime in the future, it would break

Reply:
@dr1rrb @jeromelaban Not a regression. I just checked a random commit from September and it's the same problem.

About the islands situation, I guess when we get that on gtk we can just change the condition to stop at the new equivalent of UnoGtkWindow.

Reply:
@ramezgerges is there a way to make a manual test that can help validate if this behavior is working properly? We'll need to make sure this does not regress in the future.